File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/swift/F04.P3.swift
final class Node {
    var id: Int32
    init(id: Int32) { self.id = id }
}

func probe() -> Bool {
// BEGIN PROBE F04.P3
let first = Node(id: 5)
let second = [first][0]
let same = first === second
return same
// END PROBE F04.P3
}

print(probe())

File: benchmark/2026-09-17-7677581(latest)/semantic-compression/probes/go/F04.P3.go
package main

import "fmt"

type Node struct {
	id int32
}

func probe() bool {
	// BEGIN PROBE F04.P3
	first := &Node{5}
	second := []*Node{first}[0]
	same := first == second
	return same
	// END PROBE F04.P3
}

func main() {
	fmt.Println(probe())
}
